snuba/migrations/snuba_migrations/metrics/templates.py

```diff
@@ -125,14 +125,38 @@ def get_forward_bucket_table_dist(
     retention_days
 """
 
+MATVIEW_STATEMENT_CONSOLIDATED = """
+SELECT
+    org_id,
+    project_id,
+    metric_id,
+    arrayJoin([10,60,3600,86400]) as granularity,
+    tags.key,
+    tags.value,
+    toDateTime(granularity * intDiv(toUnixTimestamp(timestamp), granularity)) as timestamp,
```

Comment:
We were not able to use `toStartOfInterval` here the way we do for the non-consolidated tables because the interval argument only accepts a constant value (it cannot be the result of an arrayJoin or another select).

This logic approximates what I did for direct aggregate writing in Python: https://github.com/getsentry/snuba/blob/a81d90790ef5b24fc4ef7a7dad90548791d21376/snuba/datasets/metrics_aggregate_processor.py#L41-L44